packages/babel-plugin-transform-es2015-arrow-functions/test/fixtures/arrow-functions/spec/expected.js

```diff
@@ -14,7 +14,7 @@ function foo() {
 
     return function () {
       babelHelpers.newArrowCheck(this, _this2);
-      return this;
```

Comment:
Up for expanding on your reasoning? My motivation was mostly that it's less separation between spec and non-spec mode, but it could be made to work either way I guess.